### ISSUE TITLE
Tests are flaky with sqlite backend

### ISSUE THREAD

**Issue by dwf** [mila-udem/blocks#942]
Both #940 and in-progress #941 have encountered failing tests with the sqlite backend. What's particularly troubling is that they are not failing deterministically, but only sometimes.

**Comment by dwf** [comment]
I've seen this one fail a number of times, again, only with sqlite.

```
======================================================================
FAIL: tests.extensions.test_training.FunctionTestCase (test_save_the_best)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/extensions/test_training.py", line 147, in test_save_the_best
    assert main_loop.log[4]['saved_to'] == (dst.name, dst_best.name)
AssertionError
```

**Comment by rizar** [comment]
As discussed today, I think we should move `SQLiteBackend`to Blocks_extras, which first and foremost requires a method of plugging additional backends from Blocks_extras to Blocks.

**Comment by dwf** [comment]
I'm going to say that we should not let the perfect be the enemy of the good, and until we can actually remove the SQLiteBackend (and figure out what is wrong with it) we should silence the failing test for the time being so that the CI status is actually useful again.

**Comment by dwf** [comment]
Superseded by #954, #955.

**Comment by dwf** [comment]
I'll let this be closed by #953 when merged.
